### ISSUE TITLE
Runtime issue: invalid pointer

### ISSUE THREAD

**Issue by japhir** [ValveSoftware/steam-for-linux#4945]
#### Your system information

* Steam client version: 1.0.0.54
* Distribution: Arch Linux 64 bit
* Opted into Steam client beta?: Yes? (cannot check without working client)
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
I expected the steam client to run. This didn't work so I deleted my preferences/settings with `rm -rf ~/.steam` and `rm ~/.steampid ~/.steampath`. When I then run steam I get the license agreement pop-up, accept it and click agree, and then get these errors: [Backtrace](https://gist.github.com/japhir/88cb449cbe3954d8bc64a95b9bf8b3ab)

#### Steps for reproducing this issue:

1. Install steam on arch linux 64 bit, with proprietary nvidia drivers
2. Run `steam-runtime` (also works with `steam` or `steam-native`) from console

It also appears that it won't upload the dump...

**Comment by kisak-valve** [comment]
Hello @japhir, does running `LANG=C steam` from a terminal improve the behavior of steam?

**Comment by japhir** [comment]
@kisak-valve hi! Unfortunately, this gives me the exact same error. [Backlog to be sure](https://gist.github.com/japhir/b7481bd2ae5f79d936d121e019b30d44)

**Comment by kisak-valve** [comment]
How about installing the `steam-native-runtime` package?

**Comment by japhir** [comment]
@kisak-valve: now it at least is successful in uploading the dump. But I get the same issue, running `LANG=C steam` or `steam-runtime` or `steam-native`.

[with dump ID](https://gist.github.com/japhir/475e3be181722848ad70dd4f97d18e2e)

**Comment by kisak-valve** [comment]
Last thing to check before I'm out of ideas. Can you check `/etc/locale.gen` and make sure the `en_US.UTF-8 UTF-8` line is not commented out (from https://bbs.archlinux.org/viewtopic.php?id=191637).

**Comment by japhir** [comment]
@kisak-valve: nope, I have `en_US.UTF-8 UTF-8`, `en_US ISO-8859-1`, and did generate my locale with `sudo locale-gen`.

**Comment by foxcm2000** [comment]
I'm also using Arch Linux and I have pretty much the exact same problem. I've messed with every locale setting I can think of and the en_US.UTF-8 UTF-8, en_US ISO-8859-1 locales are fully enabled. I even have another machine at work with a very similar configuration but it runs Steam fine.. bizarre.

I wish I had something to help you fix the problem but I've been stuck with broken Steam for the last 2 months or so.

**Comment by japhir** [comment]
So I accidentally created another issue for this (#5184), but it's the same problem five months later. Are there any developments/new ideas I could try?

I've also tried to install the steam client using flatpak (https://wiki.archlinux.org/index.php/steam) but this still didn't make the client work.

**Comment by japhir** [comment]
Seems like this has to be an issue with my `nvidia-340xx` and `nvidia-340xx-utils` video driver for my rather old Nvidia GeForce 9800 GT. 

I recently had to switch to the opensource nouveau drivers, and steam worked for a while. But because of flickering issues I had to switch back to nvidia, and now steam won't launch anymore (and I also can't launch the games independently).

**Comment by kisak-valve** [comment]
Hello @japhir, do you also have `lib32-nvidia-340xx-utils`? Does any 32 bit opengl application like `glxinfo32` from `lib32-mesa-demos` work?

**Comment by japhir** [comment]
Hello @kisak-valve. Yes, I have `lib32-nvidia-340xx-utils` installed, and the demo mesa packages (like `lib32-glxgears`) work fine.

**Comment by japhir** [comment]
Here's an example pastebin of the errors I get: https://ptpb.pw/K1nX

**Comment by japhir** [comment]
Also, I've noticed today that the [flatpak install](https://wiki.archlinux.org/index.php/Steam#Alternative_Flatpak_installation) does seem to work, but even if I add the `~/.local/share/Steam/` directory to my steam library folders, it doesn't recognize previously installed games (I have given it access to my home directory).

**Comment by kisak-valve** [comment]
Hello @japhir, are you still experiencing this issue on an up to date system? It looks like support for nvidia 340.xx has been phased out of Arch.

**Comment by japhir** [comment]
Unfortunately, the desktop system that this issue occurred on has failed, with the motherboard becoming unstable resulting in frequent, irreproducible crashes. The upside is that I'm no longer experiencing this issue on this system ;-).

**Comment by kisak-valve** [comment]
Thanks for the feedback. Closing per the last comment.

**Comment by japhir** [comment]
Hi @kisak-valve, this issue still persists for me, but now on my older laptop: 
- it's a Lenovo Thinkpad t420s with an nvidia card that I can switch to with bumblebee
- it has nvidia-390xx-dkms drivers with nvidia-390xx-utils and their lib32 counterparts
- I've tried both `steam` and `steam-native-runtime`, both result in the same segfault
- I've tried the flatpak install, which does work but there is a lot of black flickering in the client and games do not work so far

```
$> steam
Repairing installation, linking /home/japhir/.steam/steam to /home/japhir/.local/share/Steam
Running Steam on arch rolling 64-bit
STEAM_RUNTIME is enabled automatically
Pins up-to-date!
Installing breakpad exception handler for appid(steam)/version(0)
crash_20200319181002_1.dmp[46979]: Uploading dump (out-of-process)
/tmp/dumps/crash_20200319181002_1.dmp
crash_20200319181002_1.dmp[46979]: Finished uploading minidump (out-of-process): success = no
crash_20200319181002_1.dmp[46979]: error: libcurl.so: wrong ELF class: ELFCLASS64
crash_20200319181002_1.dmp[46979]: file ''/tmp/dumps/crash_20200319181002_1.dmp'', upload no: ''libcurl.so: wrong ELF class: ELFCLASS64''
/home/japhir/.local/share/Steam/steam.sh: line 906: 46972 Segmentation fault      (core dumped) $STEAM_DEBUGGER "$STEAMROOT/$STEAMEXEPATH" "$@"
```

**Comment by japhir** [comment]
On my work laptop, a Lenovo thinkpad T470s I have none of these issues. It has an integrated intel graphics card, so it may have something to do with the nvidia drivers?

**Comment by kisak-valve** [comment]
Hello @japhir, if you're using an Optimus laptop, you may have better luck trying to get Steam to run on the integrated gpu, and then figuring out how to get individual games to use the nVidia GPU on a per-game basis.

**Comment by japhir** [comment]
Thanks for the (very!) fast reply! I'm trying to launch it without primusrun/optirun. But just to check the reverse; running it with either of those also results in the same failure.
